### ISSUE TITLE
I001: `zoneinfo` not recognized as a stdlib package for PEP-8 compliant import formatting.

### ISSUE THREAD

**Issue by gtkacz** [astral-sh/ruff#13354]
Ruff doesn't recognize `zoneinfo` (introduced with PEP-615 at Python 3.9) as a first-party package and formats imports with a way not compliant with PEP-8. More details below:

- Python version: 3.12.0
- Ruff version: 0.6.5

Ruff reformats the following correct PEP-8 compliant import statement:

```python
import zoneinfo
from datetime import date, datetime, time
from typing import TYPE_CHECKING, Optional, Sequence, Tuple
```

To the following incorrect one:

```python
from datetime import date, datetime, time
from typing import TYPE_CHECKING, Optional, Sequence, Tuple

import zoneinfo
```

Based on rule `I001 [*] Import block is un-sorted or un-formatted`.

**Comment by gtkacz** [comment]
isort and Black for instance do not currently have these issues.

**Comment by AlexWaygood** [comment]
Hiya -- do you have `requires-python` or `tool.ruff.target-version` set in your `pyproject.toml` file? I can repro this in the playground without them set (which leads Ruff to infer `"py38"` as the target version, since that's our current default), but if you set `"py39"` as the target version then this isn't an issue: https://play.ruff.rs/640f351d-0d84-4638-b276-4a6239df7f20

**Comment by gtkacz** [comment]
@AlexWaygood hey, thanks for the quick response! This came up as a pre-commit check was failing on https://github.com/python-telegram-bot/python-telegram-bot/pull/4326 which does have a `requires-python = ">=3.8"` on their `pyproject.toml` file. They do plan on dropping support on Python 3.8 soon, so I believe that will fix this, but even then, shouldn't Ruff properly format imports without a `pyproject.toml` file set-up? Thanks for your time! :)

**Comment by AlexWaygood** [comment]
Well, the issue here is that we assume you'll only be importing stdlib modules available on all versions of Python you support. If you support Python 3.8, we'll only look at the list of stdlib modules available on Python 3.8 when trying to determine if a module is a stdlib module or a third-party module. So I *think* Ruff's behaviour is correct here :-)

If this is causing issues for you, you might find this setting helpful: https://docs.astral.sh/ruff/settings/#lint_isort_extra-standard-library

**Comment by gtkacz** [comment]
@AlexWaygood that worked for the pre-commit issue, thanks :)! Still, I think if `pyproject.toml` isn't used at all, the imports should be properly formatted. Maybe include a default value of

```toml
[tool.ruff.lint.isort]
extra-standard-library = ["zoneinfo"]
```

When it's not overridden if `requires-python` isn't passed? Specially since the `isort` hook is properly formatting with the same `requires-python = ">=3.8"` configuration in the PR I mentioned

**Comment by AlexWaygood** [comment]
Hmm, I guess I'm not sure why you'd be running two separate import sorters in CI (Ruff and isort)! Just one should probably do the trick 😄

That also honestly sounds like a bug in isort rather than ruff if they're treating `zoneinfo` as a stdlib module on Python versions before it was added to the stdlib.

> Still, I think if `pyproject.toml` isn't used at all, the imports should be properly formatted.

We also support the same configuration options via a `ruff.toml` file, if you don't have a `pyproject.toml` file! Also, if you want to tell ruff not to infer the target version from the `requires-python` field, you can set the `tool.ruff.target-version` option instead :-) https://docs.astral.sh/ruff/settings/#target-version

**Comment by gtkacz** [comment]
Don't ask me on this one lol 🤷 thanks for the help anyway! :)
